client/dashboard/src/contexts/Auth.tsx

```diff
@@ -231,7 +232,14 @@ const AuthHandler = ({ children }: { children: React.ReactNode }) => {
       preferredProject = session.organization.projects[0]!.slug;
     }
 
-    navigate(`/${session.organization.slug}/${preferredProject}`);
+    const paramsToForward = [LINKED_FROM_PARAM];
+    const paramsToForwardString = paramsToForward
+      .map((p) => `${p}=${searchParams.get(p)}`)
+      .join("&");
```

Comment:
Not sure what `LINKED_FROM_PARAM` will be but using [URLSearchParams](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) is the safer approach as it will handle encoding. A few places in the PR could benefit from this.